ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/ChainMethodContinuationTest.kt

```diff
@@ -0,0 +1,172 @@
+package com.pinterest.ktlint.ruleset.standard.rules
+
+import com.pinterest.ktlint.test.KtLintAssertThat.Companion.assertThatRule
+import com.pinterest.ktlint.test.LintViolation
+import org.junit.jupiter.api.Test
+
+class ChainMethodContinuationTest {
+    private val chainMethodContinuationAssertThat = assertThatRule { ChainMethodContinuation() }
+
+    @Test
+    fun `Given method chain which is correctly wrapped in single line`() {
+        val code =
+            """
+            val foo = listOf(1, 2, 3).filter { it > 2 }!!.takeIf { it.count() > 100 }.map { it * it }?.sum()!!
+            """.trimIndent()
+        chainMethodContinuationAssertThat(code).hasNoLintViolations()
+    }
+
+    @Test
+    fun `Given method chain which is incorrectly wrapped`() {
+        val code =
+            """
+            val foo = listOf(1, 2, 3).
+                filter { it > 2 }!!.
+                takeIf { it.count() > 100 }.
+                map {
+                    it * it
+                }?.
+                map {
+                    it * it
+                }!!.
+                sum()
+            """.trimIndent()
+        val formattedCode =
+            """
+            val foo = listOf(1, 2, 3)
+                .filter { it > 2 }!!
+                .takeIf { it.count() > 100 }
+                .map {
+                    it * it
+                }?.map {
+                    it * it
+                }!!.sum()
+            """.trimIndent()
+        chainMethodContinuationAssertThat(code)
+            .hasLintViolations(
+                LintViolation(1, 27, ". must merge at the start of next call"),
+                LintViolation(2, 25, ". must merge at the start of next call"),
+                LintViolation(3, 33, ". must merge at the start of next call"),
+                LintViolation(6, 8, "}?. must merge at the start of next call"),
+                LintViolation(9, 9, "}!!. must merge at the start of next call"),
+            ).isFormattedAs(formattedCode)
+    }
+
+    @Test
+    fun `Given method with RBRACE is not wrapped with next chain operator`() {
```

Comment:
Just out of curious. Are you looking forward from the RBRACE to see whether it is followed by a method chain or are you looking backward if the DOT or SAFE_ACCESS is preceded by a dangling RBRACE? Usually it is better to look back instead of forward as all nodes that have been processed before are already cleaned up if needed, nodes after the current node are not yet cleaned.

And please split into smaller tests.

Reply:
Coincidentally I am looking backwards. But I was not aware of this. My code to find the brace is
```
val leftExpressionEndBrace =
    chainOperator
        .prevLeaf { !it.isWhiteSpace() && it.elementType != EXCLEXCL }
        ?.lastChildLeafOrSelf() ?: return
val leftExpressionBeforeEndBrace = leftExpressionEndBrace.prevCodeSibling()?.lastChildLeafOrSelf() ?: return
val isLeftExpressionEndBraceInSeparateLine =
    leftExpressionEndBrace.elementType == RBRACE &&
        hasNewLineInClosedRange(leftExpressionBeforeEndBrace, leftExpressionEndBrace)
```

Reply:
Given that you're lookingbackward, you can also reflect this in the name of the. Given a chain operator on a separate line below the rbrace of a multiline chain.

Reply:
I have updated the verbiage a little bit also making it more consistent